File: chatRoom.cpp
#include<iostream>
using namespace std;

int main (){
	string s;
	cin >> s;

	string standard = "hello";
	int idx = 0;
	int len = s.length();
	bool found =  false;

	for(int i=0;i<len;i++){

		if(s[i] == standard[idx]) idx++;
		// else{
		// 	if(s[i] == standard[idx -1]) continue;
		// 	else idx = 0;
		// }

		if(idx == standard.length()){
			found = true;
			break;
		}
	}

	if(found){
		cout << "YES";
	}else{
		cout << "NO";
	}
	return 0;
}

File: chatRoom.js
var chatRoom =(s)=>{
	var standard = 'hello';
	var idx = 0;
	var len = s.length;

	for(var i=0;i < len;i++){

		if(s[i] == standard[idx]) idx ++;
		// else {
		// 	if(s[i] == standard[idx -1]) continue;
		// 	else idx = 0;
		// }

		if(idx == standard.length) return 'YES';
	}

	return 'NO';
}

console.log(chatRoom('ahhellllloou'));// YES
console.log(chatRoom('hlelo')) //NO

console.log(chatRoom('ahhellaallloou'))//NO